Shared/Components/ImageView.swift

```diff
@@ -66,6 +69,10 @@ struct ImageView: View {
                 }
             }
             .pipeline(imagePipeline)
+            .opacity(isImageLoaded ? 1 : 0)
```

Comment:
Sorry, but this still isn't correct and is also a big change in behavior. This just fades in the entire `ImageView` on appearance (which should probably actually happen with a `.transition`) and isn't what we want.

I assume you're working with a local server so that images are loading quite fast, which means that the lack of transition between the placeholder blur hash images and the final image isn't noticeable. With a server that takes a few seconds to load an image, the lack of a fade is still noticeable.

I took some time to take a look at this to help point to what the problem actually is, which is one I assumed in my last comment:

> size differences between the blur hash view and the final image view.

Here are steps below that cause the issue:
- For these header views, the aspect ratio is defined to `.fill`, which will cause the `ImageView` to aspect fill its container. Note: `ImageView` will have the same size of its contents (placeholder, failure, image).
	- https://github.com/jellyfin/Swiftfin/blob/384e80805e6c8949e29f67a4324e005d70602c72/Swiftfin/Views/ItemView/iOS/ScrollViews/CompactLogoScrollView.swift#L32

- For the placeholder view, this would create a size that matched the height of the defined header and with a width of the header, for example: `width: 393, height: 393`.
	- https://github.com/jellyfin/Swiftfin/blob/384e80805e6c8949e29f67a4324e005d70602c72/Shared/Extensions/ViewExtensions/Modifiers/BackgroundParallaxHeaderModifier.swift#L42

- When the image finally loads, it will also aspect fill the header but will get a different size, for example: `width: 530, height: 393` (the heights will always be the same due to rules of aspect fill)
- Since the `ImageView` will have the same size of its contents, and its contents changed, it needs to resize and recenter itself. That behavior is the "slide" that we see.

### Solution

The solution is to make the `ImageView` always have the same aspect ratio. If all of the views have the same aspect ratio - which means that they will size the same and correctly. This is used on the `.posterStyle` modifier to make all the posters, which is kind of why we don't see any "slide" behavior with them.

If you want to see this behavior yourself, I would recommend adding the modifier below in some places (like on the placeholder and image in `ImageView` or the header in each item view) to see the size differences before and after the fix:

```
.onSizeChanged { size in
	print(size)
}
```

### Merge

I did the work on my branch below, as well as left a `TODO` for some polish regarding the entire transition between a blur hash and the image view. You are welcome to merge that branch into this and I can approve or I can make the PR instead.

- https://github.com/jellyfin/Swiftfin/compare/main...LePips:SwiftFin:fix-item-header-image-slide

Reply:
Ah ok, I was thinking it felt kinda different but wasn't sure. Yeah my server is local so everything is pretty instant. 

Ah that makes sense. Thanks for the explanation, much appreciated.

I'll merge into mine and work on the todo.

Reply:
That `TODO` doesn't need to be done immediately and is just a UX polish. That may actually take a lot of work/a refactor and may take a lot of work.

Reply:
Ah ok. I'll have a look anyway